### PR TITLE
Update deepac to 0.13.2

### PR DESCRIPTION
**Pull request by BiocondaBot**
Update [`deepac`](https://bioconda.github.io/recipes/deepac/README.html): **0.12.2** &rarr; **0.13.2**

[install with bioconda](http://bioconda.github.io/recipes/deepac/README.html) [Conda](https://anaconda.org/bioconda/deepac/files)

Info | Link
-----|-----
Recipe | [`recipes/deepac`](https://github.com//bioconda/bioconda-recipes/tree/bump/deepac/recipes/deepac) (click to view/edit other files)
Releases |[https://pypi.org/pypi/deepac/json](https://pypi.org/pypi/deepac/json)
Recipe Maintainer(s) | @JakubBartoszewicz

***

This pull request was automatically generated (see [docs](https://bioconda.github.io/contributor/updating.html)).

### CODE REVIEW

**Comment by JakubBartoszewicz** [comment]
Hi @dpryan79, thanks for help! I seem to have some problems - I get a segfault in the mulled test when it tries to run the biocontainer... Do you know what could be wrong by any chance?

**Comment by dpryan79** [comment]
@JakubBartoszewicz I'm not really sure what the problem is, perhaps some sort of memory limitation on the CI nodes? I wouldn't expect loading a python module to cause that though. There's no non-python code here, correct?

**Comment by JakubBartoszewicz** [comment]
Not at all, just python. It also fails in a local test (in the same way). The differences compared to the previous versions shouldn't affect memory, at least not for the test commands used in the recipe.

I tested the recipe for the previous version (0.12.2) locally* and it actually crashes with the same segfault (even though it used to work, ofc)! weird...

*Caveat: some dependencies introduced breaking changes so I had to slightly modify it to make it work as it used to; I'll update the recipe build as soon as the segfault problem is solved

**Comment by JakubBartoszewicz** [comment]
I noticed that requiring python<3.8 helps locally (all tests pass, including the mulled container). I don't know why yet (the new version was developed and tested using python 3.8). Unfortunately, the CI tests fail anyway.

**Comment by JakubBartoszewicz** [comment]
Dear @dpryan79, @bioconda/core,

sorry to bug you, but could I ask you for help? I'm getting a segfault at the end of the mulled test even though it's just a python package and the help message gets printed out successfully. In all other cases everything works. I tried to find the reason for that  for quite a long time but I'm out of ideas - there's not much info in the error message... I really care about keeping the up-to-date version on bioconda, but I hit the wall.

I suspect this is not caused by the package itself or by the recipe itself (the reason: tests failed also for an older version of the recipe that passed everything without problems back in July). But what is it then - I don't know.

Could you please advise what I could do? I assume disabling the mulled test is not an option...?

I'll be grateful for any tips or suggestions.

**Comment by dpryan79** [comment]
@JakubBartoszewicz I'll try to run this locally and save the container later if no one else gets around to it first.

**Comment by mbargull** [comment]
Yeah, I don't understand this. I created an env locally and ran `deepac --help` in the same base container but couldn't reproduce the segfault. This is a really odd one...

**Comment by JakubBartoszewicz** [comment]
@dpryan79, @mbargull thank you both. I removed the python<3.8 requirement since the package supports it and it was there only because it caused local tests to pass... which no longer seems to be the case. It doesn't help but removes an unnecessary requirement.

File: recipes/deepac/meta.yaml [hunk 1]
```diff
@@ -44,7 +48,7 @@ test:
     - deepac.explain
     - deepac.gwpa
   commands:
-    - deepac --help
+    - deepac --version | grep -qF '{{ version }}'
```
Comment (mbargull) [inline recipes/deepac/meta.yaml:51]: Let's fetch the artifacts after the build so we can test the container from the CI.

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Comment by mbargull** [comment]
(Triggered with imports from `/usr/local/lib/python3.8/site-packages/deepac/eval/eval.py `)
```sh
from tensorflow.keras import backend
import matplotlib.pyplot as plt
```
segfaults -- but only on the first run.
... Needs some more investigation...

**Comment by dpryan79** [comment]
Wow, that's...confusing.

**Comment by mbargull** [comment]
This reliably reproduces it in the container:
```
rm -f /root/.cache/matplotlib/fontlist-v330.json ; python -c 'from tensorflow.keras import backend;import matplotlib.pyplot as plt'
```

**Comment by JakubBartoszewicz** [comment]
@mbargull I have no idea how you found that but that's awesome. Thanks.

Import odrering matters - this works:
`rm -f /root/.cache/matplotlib/fontlist-v330.json ; python -c 'import matplotlib.pyplot as plt;from tensorflow.keras import backend'`

**Comment by JakubBartoszewicz** [comment]
I released a patch reordering the imports and importing matplotlib at the beggining of the entry point file. It passed local tests, if it also passes the CI tests I guess it's safe to merge.

Without importing matplotlib at the beginning of the entry point file, I was getting a more detailed error info. I'm pasting it here for future reference (it should not appear in the new patch anymore).

```
19:57:27 BIOCONDA INFO (ERR) [Nov 19 19:57:27] DEBU Container [fd68005f7180 step-7b072a90aa] started, waiting for completion
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR XXX lineno: 278, opcode: 255                                                                                                                             
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR Traceback (most recent call last):                                                                                                                       
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/bin/deepac", line 6, in <module>                                                                                                        
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR from deepac.command_line import main                                                                                                                     
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/site-packages/deepac/command_line.py", line 18, in <module>                                                               
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR from deepac.eval.eval import evaluate_reads                                                                                                              
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/site-packages/deepac/eval/eval.py", line 8, in <module>                                                                   
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR import matplotlib.pyplot as plt                                                                                                                          
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/site-packages/matplotlib/pyplot.py", line 36, in <module>                                                                 
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR import matplotlib.colorbar                                                                                                                               
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/site-packages/matplotlib/colorbar.py", line 44, in <module>                                                               
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR import matplotlib.contour as contour                                                                                                                     
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/site-packages/matplotlib/contour.py", line 21, in <module> 
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR import matplotlib.texmanager as texmanager                                                                                                       
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/site-packages/matplotlib/texmanager.py", line 47, in <module>                                                             
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR class TexManager:                                                                                                                                        
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/site-packages/matplotlib/texmanager.py", line 230, in TexManager                                                          
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR _re_vbox = re.compile(
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/re.py", line 252, in compile
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR return _compile(pattern, flags)
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/re.py", line 304, in _compile
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR p = sre_compile.compile(pattern, flags)
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/sre_compile.py", line 768, in compile
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR code = _code(p, flags)
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/sre_compile.py", line 607, in _code
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR _compile(code, p.data, flags)
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/sre_compile.py", line 168, in _compile
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR _compile(code, p, _combine_flags(flags, add_flags, del_flags))
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/sre_compile.py", line 148, in _compile
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR _compile(code, av[2], flags)
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/sre_compile.py", line 120, in _compile
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR charset, hascased = _optimize_charset(av, iscased, tolower, fixes)
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR File "/usr/local/lib/python3.8/sre_compile.py", line 278, in _optimize_charset
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR out = []
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR SystemError: unknown opcode
19:57:29 BIOCONDA INFO (ERR) [Nov 19 19:57:29] SERR Segmentation fault (core dumped)
```

**Comment by JakubBartoszewicz** [comment]
@mbargull, @dpryan79 thank you so much for your help. It seems to work now!

**Review by JakubBartoszewicz — APPROVED** [review]

**Comment by dpryan79** [comment]
This has got to be the craziest cause of a segfault that I've seen this year. As it's very obviously not due to anything in deepac I have no objections to merging this. Thoughts @mbargull?

**Comment by mbargull** [comment]
Well, it's just hard to tell what the underlying cause is. There are too many hacky/non-optimal parts at play (upstream)...
We can merge and hope the workaround is sufficient. Otherwise, I'd assume it can take a couple of hours to find the root case and depending on what it is, even more time to fix it... :/
@JakubBartoszewicz, if you're okay with a little bit of crossing fingers, feel free to `@BiocondaBot, please merge` this.

**Comment by JakubBartoszewicz** [comment]
Thanks. Fingers crossed!

@BiocondaBot please merge

**Comment by JakubBartoszewicz** [comment]
@BiocondaBot please merge

**Pull request merged**